doc/man7/proxy-certificates.pod: New guide for proxy certificates

levitte opened the pull request:
This replaces doc/HOWTO/proxy_certificates.txt

levitte committed:
doc/man7/proxy-certificates.pod: New guide for proxy certificates

This replaces doc/HOWTO/proxy_certificates.txt

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+=item *
+
+They must not have the extensions B<subjectAltName> or
+B<issuerAltName>.
```
Move ‚extensions‘ to the end of the sentence. (That’s the correct ordering according to my limited understanding of English grammar.)

levitte replied:
As far as I know, this is correct English. @mattcaswell?

paulidale replied:
@mspncp's suggestion seems the more natural to me.  Alternatively, add a colon:
> They must not have the extensions: **subjectAltName** or **issuerAltName**.

levitte replied:
It seems weird to me, or ambiguous at best, as "the **subjectAltName** extension" could mean that **subjectAltName** has an extension (not named here).

Regarding noun before or after a proper name (which is what we're disputing), it seems like it's hard to get an absolute agreement, but from reading the following, I'm still leaning toward what I wrote:

https://english.stackexchange.com/questions/114871/does-a-name-go-before-or-after-the-noun-it-modifies

levitte replied:
The suggest colon form mostly feel clumsy.  Slightly better if "the" is replaced with "these":

> They must not have these extensions: **subjectAltName** or **issuerAltName**.

levitte replied:
Side note: *never* had I thought that working with OpenSSL would lead me to have *English language disputes*.  Oy...

richsalz replied:
"must not have the subjectAltName or issuerAltName extensions" is the best way to say this.

paulidale replied:
> It seems weird to me, or ambiguous at best, as "the **subjectAltName** extension" could mean that subjectAltName has an extension (not named here).

This interpretation would usually be expressed using the genitive: "the **subjectAltName**'s extension" (equally: "the extension of the **subjectAltName**").  People, being lazy, would forget.

mattcaswell replied:
Either form sounds correct to me, but @mspncp's suggestion sounds more natural and so I would prefer that form.

levitte replied:
...

Ok then, I give in

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+
+=item *
+
+They must have the extension B<proxyCertInfo>.
```
Same here.

richsalz replied:
Must have the proxyCertInfo extension.

mspncp replied:
> They must not have the extensions B\<subjectAltName\> or
B\<issuerAltName\>.

In German language this would be the correct ordering, too, so I can understand your reluctance. I still believe that in English it's other way around , but I won't insist on it. I'll just content myself with the fact that two native speakers voted in favour of my suggestion ;-)

paulidale replied:
@richsalz speaks American, I speak Australian.
Neither is quite English.

mspncp replied:
It surprises me that you consider American and Australian English not to be proper English ;-)

levitte replied:
I doubt that this issue depends on any geographical variant of Standard English.  From having looked around, I have seen no trace of British vs TheRestOfTheAnglosphere dispute on this subject, so...

paulidale replied:
I deliberately avoided the word _proper_ :smiley:

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+
+=item B<text>
+
+indicates that the string is simply bytes, without any encoding:
```
is simply bytes -> is a byte sequence

levitte replied:
Fixed

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+
+=item B<hex>
+
+indicates the string is encoded in hex, with colons between each byte
```
encoded in hex  - hexadecimal encoded binary data

levitte replied:
Done

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+
+=item B<file>
+
+indicates that the text of the policy should really be taken from a
```
Remove ‚really‘ (twice)

levitte replied:
Done

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+=back
+
+I<NOTE: The proxy policy value is what determines the rights granted
+to the process uring the proxy certificate.  It's up to the
```
uring?

levitte replied:
Fixed

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+        -extfile proxy.cnf -extensions v3_proxy1
+
+You can also create a proxy certificate using another proxy
+certificate as issuer (note: I'm using a different configuration
```
I ?

levitte replied:
Rephrased a bit

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+=head2 Using proxy certs in applications
+
+To interpret proxy policies, the application would normally start with
+some default rights (perhaps none at all), then compute the resulting
```
rights  - permissions?

levitte replied:
Proxy certificates are usually discussed in terms of rights

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+=item *
+
+a callback function that will be called for every certificate being
+validated.  The callback be called several times for each certificate,
```
be called-> will be called

levitte replied:
Almost.  Fixed

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+
+To this date, it seems that proxy certificates have only been used in
+a context highly aware of them, and no one seems to have investigated
+how they can be used of misused outside of such an environment.
```
of misused -> or misused

levitte replied:
Thanks

mspncp commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+=head1 NOTES
+
+To this date, it seems that proxy certificates have only been used in
+a context highly aware of them, and no one seems to have investigated
```
Remove ‚highly‘?

levitte replied:
Yup, and rephrased slightly

mspncp reviewed (commented):
Just some minor corrections and wording suggestions. 

Did not check the example source code. Does it compile?

richsalz commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+=head1 DESCRIPTION
+
+Proxy certificates are defined in
+L<RFC 3820|https://tools.ietf.org/html/rfc3820>.  They are used to
```
I believe all other RFC references just have the link in the SEE ALSO section.

levitte replied:
Ah, thanks.  I intended to do exactly that, but forgot.

richsalz replied:
Oh and also, the official RFC link is https://www.rfc-editor.org/rfc/rfcXXXX.html

levitte replied:
Actually, this manpage is almost pioneering that.  The only other that I could find was the tsget manual.

richsalz replied:
One of my pending PR's fixes tsget :)

levitte commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,359 @@
+L<RFC 3820|https://tools.ietf.org/html/rfc3820>.  They are used to
+extend rights to some other entity (a computer process, typically, or
+sometimes to the user itself).  This allows the entity to perform
+operations on behalf of the owner of the EE certificate.
```
Done

levitte committed:
fixup! doc/man7/proxy-certificates.pod: New guide for proxy certificates

levitte committed:
util/find-doc-nits: Better parsing of links

When checking links, we need to peal away stuff that aren't part of
the link proper first.  That makes it easier to check the link
itself.

mspncp reviewed (commented):
LGTM now, my suggestions have (almost) all been addressed.  Since I did not go into the details of the RFC and the perl code changes, I won't approve formally.

levitte committed:
fixup! doc/man7/proxy-certificates.pod: New guide for proxy certificates

levitte commented:
All review comments have been addressed

paulidale commented on doc/man7/proxy-certificates.pod:
```diff
@@ -0,0 +1,362 @@
+    static int get_proxy_auth_ex_data_idx(X509_STORE_CTX *ctx)
+    {
+        static volatile int idx = -1;
+        if (idx < 0) {
```
Blank line after declaration.

levitte replied:
Fixed

paulidale reviewed (approved):
Seems good, although I'm not well versed in the various standards involved.
One nit.

paulidale commented:
The formality: good for 1.1.1 and master.

levitte committed:
fixup! doc/man7/proxy-certificates.pod: New guide for proxy certificates

levitte commented:
Merged

master:
2a03823606b2d7e73e5dc890c3202e186511151f doc/man7/proxy-certificates.pod: New guide for proxy certificates
76fde1db3c2c2159aaf56fb8995368c25f88f9a4 util/find-doc-nits: Better parsing of links

1.1.1:
40d422fd3a685c3e1e62ab37eda1189567d000b5 doc/man7/proxy-certificates.pod: New guide for proxy certificates

Note that the util/find-doc-nits change didn't get cherry-picked, because the foundation for the change isn't there in 1.1.1.
